playground/index.html

```diff
@@ -211,10 +220,23 @@
     output.setFontSize(`${fontSize}px`)
     logOutput.setFontSize(`${fontSize}px`)
   }
+
+  function generateShareLink() {
+    const compressedDataBase64 = LZString.compressToEncodedURIComponent(JSON.stringify(
+      {
+        "sourceCode": editor.getValue(),
+        "options": re2cFlagsDropBox.value,
+        "language": languageSelect.options[languageSelect.selectedIndex].text,
+      }
+    ));
+    const shareableURL = `${window.location.origin}${window.location.pathname}?compressed=${compressedDataBase64}`;
+    navigator.clipboard.writeText(shareableURL)
+    window.alert("The share link is copied to clipboard");
```

Comment:
This didn't work for me on localhost, presumably because of a non-secure context (the error was `navigator.clipboard is undefined`). I was able to fix it by replacing the highlighted code with this:
```js
    if (navigator.clipboard && window.isSecureContext) {
        navigator.clipboard.writeText(shareableURL)
    } else {
        // Use the 'out of viewport hidden text area' trick
        const textArea = document.createElement("textarea")
        textArea.value = shareableURL

        // Move textarea out of the viewport so it's not visible
        textArea.style.position = "absolute"
        textArea.style.left = "-999999px"

        document.body.prepend(textArea)
        textArea.select()

        try {
            document.execCommand('copy')
        } catch (error) {
            console.error(error)
            alert("Cannot copy shared link!")
        } finally {
            textArea.remove()
        }
    }
    window.alert("Share link is copied to clipboard.")
```

Reply:
Oh god :)
I have fixed it in a better way. Please check that it works on your machine.

Reply:
Yes, it does. Thanks.